instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -84,7 +84,7 @@ def span_attrs(kind, *args) # rubocop:disable Metrics/AbcSize
             end
 
             attrs = { 'db.operation' => validated_operation(operation), 'db.postgresql.prepared_statement_name' => statement_name }
-            attrs['db.statement'] = sql unless config[:db_statement] == :omit
+            attrs['db.statement'] = sql.to_s if config[:db_statement] != :omit && !sql.nil? && sql.respond_to?(:to_s)
```

Comment:
While I do ❤️ defensive programming in general, I'm wondering if we need to be _quite_ this defensive. For my own edification, I'm wondering what scenarios we're trying to avoid?

To explain why I'm curious - anything inheriting from `Object` should always have `to_s` defined, so only things explicitly inheriting from `BasicObject` would fail the `respond_to?` check. And, even if `sql` is `nil` (which _could_ happen, depending on how the interactions with the `lru_cache` go) - `to_s` is still defined, although admittedly a blank string as an attribute value seems useless.

It's totally fine either way and I'm approving the PR regardless, but I was just curious what the logic was behind the defensiveness.

Reply:
the nil check is to prevent `""` as an attribute key. agreed the  `to_s` is probably unnecessary. will update

Reply:
👍 Cool, thanks for clarifying. I do appreciate it! 😄